infrastructure-stack-database-props.ts

```diff
@@ -0,0 +1,89 @@
+import { InstanceType } from "aws-cdk-lib/aws-ec2";
+import { PostgresEngineVersion } from "aws-cdk-lib/aws-rds";
+import { Duration } from "aws-cdk-lib";
+
+/**
+ * A regular Postgres instance database type we can
+ * ask to be installed as part of our infrastructure
+ */
+export type PostgresInstance = PostgresCommon & {
+  type: "postgres-instance";
+
+  instanceType?: InstanceType;
+
+  /**
+   * If set will override the postgres engine used - otherwise
+   * we will make this by default aggressively track the latest postgres release
+   */
+  overridePostgresVersion?: PostgresEngineVersion;
```

Comment:
I think this could also be a common option. E.g. to override the postgres version for serverless?